core/src/main/java/io/questdb/cairo/TxnScoreboardV2.java

```diff
@@ -130,10 +145,20 @@ public long getMin() {
         }
 
         long min = Long.MAX_VALUE;
-        for (int i = 0; i < entryScanCount; i++) {
-            long lockedTxn = Unsafe.getUnsafe().getLongVolatile(null, entriesMem + (long) i * Long.BYTES);
-            if (lockedTxn > UNLOCKED) {
+        for (int i = 0; i < bitmapCount; i++) {
+            long bitmap = Unsafe.getUnsafe().getLongVolatile(null, bitmapMem + (long) i * Long.BYTES);
+            if (bitmap == 0) {
+                continue;
+            }
+
+            int base = i * Long.SIZE;
+            while (bitmap != 0) {
+                final long lowestBit = Long.lowestOneBit(bitmap);
+                final int bit = Long.numberOfTrailingZeros(lowestBit);
+                int internalId = base + bit;
+                long lockedTxn = Unsafe.getUnsafe().getLongVolatile(null, entriesMem + (long) internalId * Long.BYTES);
                 min = Math.min(min, lockedTxn);
```

Comment:
This assignment should be kept in a `if (lockedTxn > UNLOCKED)` check like in the original code. Otherwise, with concurrent acquire/release calls, this method may wrongly return `UNLOCKED` value.